### ISSUE TITLE
is tf.signal.stft same as librosa.stft ?

### ISSUE THREAD

**Issue by ucasiggcas** [tensorflow/tensorflow#32373]
hi,
Dear, if I use the tf.signal.stft instead of librosa.stft, will have any difference ?
and tf.signal.inverse_stft could be librosa.istft ?

thanks a lot

**Comment by jvishnuvardhan** [comment]
This is not Build/Installation or Bug/Performance issue. Please post this kind of support questions at [Stackoverflow](https://stackoverflow.com/questions/tagged/tensorflow). There is a big community to support and learn from your questions. GitHub is mainly for addressing bugs in installation and performance. Thanks!

**Comment by ucasiggcas** [comment]
> Are you satisfied with the resolution of your issue?
> [Yes](https://docs.google.com/forms/d/e/1FAIpQLSfaP12TRhd9xSxjXZjcZFNXPGk4kc1-qMdv3gc6bEP90vY1ew/viewform?entry.85265664=Yes&entry.2137816233=32373)
> [No](https://docs.google.com/forms/d/e/1FAIpQLSfaP12TRhd9xSxjXZjcZFNXPGk4kc1-qMdv3gc6bEP90vY1ew/viewform?entry.85265664=No&entry.2137816233=32373)

NO
Can't open the link

**Comment by carlthome** [comment]
They're different and you cannot just replace one with the other. This is pretty easy to test.

```py
import librosa as lr
import tensorflow as tf

waveform, samplerate = lr.load(lr.util.example_audio_file(), None, duration=10.0)

z1 = lr.stft(waveform)
z2 = tf.signal.stft(waveform, 2048, 512).numpy()
print(z1.shape, z1.min(), z1.max(), z1.mean())
print(z2.shape, z2.min(), z2.max(), z2.mean())
```

Aside from the transpose, librosa does reflect padding to center analysis frames (`np.pad(waveform, 2048//2, 'reflect')`). That seems to be the major difference, but even after matching that you'd still get some slightly different values (perhaps the underlying FFT isn't identical). They're different enough that you cannot just drop-in replace one for the other with a pretrained model so beware.
